test/Garnet.test/RespConfigTests.cs

```diff
@@ -0,0 +1,241 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Linq;
+using Garnet.common;
+using Garnet.server;
+using NUnit.Framework;
+using NUnit.Framework.Legacy;
+using StackExchange.Redis;
+
+namespace Garnet.test
+{
+    [TestFixture]
+    public class RespConfigTests
+    {
+        GarnetServer server;
+        private string memorySize = "17g";
+        private string indexSize = "64m";
+        private string objectStoreLogMemorySize = "17m";
+        private string objectStoreHeapMemorySize = "32m";
+        private string objectStoreIndexSize = "8m";
+
+        [SetUp]
+        public void Setup()
+        {
+            TestUtils.DeleteDirectory(TestUtils.MethodTestDir, wait: true);
+            server = TestUtils.CreateGarnetServer(TestUtils.MethodTestDir,
+                memorySize: memorySize,
+                indexSize: indexSize,
+                objectStoreLogMemorySize: objectStoreLogMemorySize,
+                objectStoreIndexSize: objectStoreIndexSize,
+                objectStoreHeapMemorySize: objectStoreHeapMemorySize);
+            server.Start();
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            server.Dispose();
+            TestUtils.DeleteDirectory(TestUtils.MethodTestDir);
+        }
+
+        [Test]
+        [TestCase(true, "16g", "32g", "64g", "g4")]
+        [TestCase(true, "9gB", "28GB", "33G", "2gBB")]
+        [TestCase(false, "16m", "32m", "64m", "3bm")]
+        [TestCase(false, "5MB", "30M", "128mb", "44d")]
+
+        public void ConfigSetMemorySizeTest(bool mainStore, string smallerSize, string largerSize, string largerThanBufferSize, string malformedSize)
```

Comment:
Can use `var info = TestUtils.GetStoreAddressInfo(server)` to get addresses (see `PersistCopyUpdateTest`) and verify that after expanding the memory size, HeadAddress does not move forward when sufficient new records are inserted at tail (because the log size is made larger).